bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java

```diff
@@ -232,31 +240,44 @@ private void updateZoneAlarm(WhatAlarm w) {
         switch (w) {
             case ZONE_ALARM_INTRUSION:
                 updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_INTRUSION));
+                updateZoneAlarmTimeStamp();
```

Comment:
you have to choose when updateZoneAlarmTimeStamp is called. This way it's called 2 times: in the switch inside updateZoneAlarm and then again in the updateZone switch

Reply:
You are right, I  choose to call it in updateZone() to invoke it only once

Reply:
done